iree/compiler/Dialect/Stream/Transforms/DumpStatistics.cpp

```diff
@@ -0,0 +1,555 @@
+
+  // Globals:
+  os << llvm::formatv("{0},{1},{2},{3},", stats.constantCount,
+                      stats.constantSize, stats.variableCount,
+                      stats.variableSize);
+
+  // Synchronization:
+  os << llvm::formatv("{0},", stats.awaitCount);
+
+  // Execution:
+  os << llvm::formatv("{0},{1},{2},{3},{4},", stats.submissionCount,
+                      stats.transientSize, stats.fillCount, stats.copyCount,
+                      stats.dispatchCount);
+
+  // Executables:
+  os << llvm::formatv("{0}", stats.executableCount);
+
+  os << "\n";
+  os << "\n";
+}
+
+static void dumpExecutionCSVTable(const UsageInfo &usageInfo,
+                                  IREE::Stream::CmdExecuteOp executeOp,
+                                  llvm::raw_fd_ostream &os) {
+  os << "; ";
+  prettyPrintOpBreadcrumb(executeOp, os);
+  os << "\n";
+  os << R"("Depth","Command","Symbol","Length","Invocations","X","Y","Z","Operands","Resources")";
+  os << "\n";
+  std::function<void(Operation *)> dumpRow;
+  int depth = 0;
+  dumpRow = [&](Operation *op) {
+    TypeSwitch<Operation *>(op)
+        .Case<IREE::Stream::CmdSerialOp>([&](auto op) {
+          ++depth;
+          for (auto &nestedOp : op.body().front()) dumpRow(&nestedOp);
+          --depth;
+        })
+        .Case<IREE::Stream::CmdConcurrentOp>([&](auto op) {
+          ++depth;
+          for (auto &nestedOp : op.body().front()) dumpRow(&nestedOp);
+          --depth;
+        })
+        .Case<IREE::Stream::CmdFillOp>([&](auto op) {
+          APInt length;
+          matchPattern(op.target_length(), m_ConstantInt(&length));
+          os << llvm::formatv(R"({0},"fill",,{1},,,,,,)", depth, length);
+          os << "\n";
+        })
+        .Case<IREE::Stream::CmdCopyOp>([&](auto op) {
+          APInt length;
+          matchPattern(op.length(), m_ConstantInt(&length));
+          os << llvm::formatv(R"({0},"copy",,{1},,,,,,)", depth, length);
+          os << "\n";
+        })
+        .Case<IREE::Stream::CmdDispatchOp>([&](auto op) {
+          auto workload = op.workgroup_count();
+          APInt workloadX;
+          APInt workloadY;
+          APInt workloadZ;
+          matchPattern(workload[0], m_ConstantInt(&workloadX));
+          matchPattern(workload[1], m_ConstantInt(&workloadY));
+          matchPattern(workload[2], m_ConstantInt(&workloadZ));
+          os << llvm::formatv(
+              R"({0},"dispatch","{1}",,{2},{3},{4},{5},{6},{7})", depth,
+              op.entry_point(), workloadX * workloadY * workloadZ, workloadX,
+              workloadY, workloadZ, op.operands().size(),
+              op.resources().size());
+          os << "\n";
+        });
+  };
+  for (auto &op : executeOp.body().front()) {
+    dumpRow(&op);
+  }
+  os << "\n";
+}
+
+static void dumpCSVTables(const UsageInfo &usageInfo,
+                          llvm::raw_fd_ostream &os) {
+  os << ";\n";
+  os << "; Aggregate Statistics (static, whole-program)\n";
+  os << ";\n\n";
```

Comment:
gists don't seem to like this csv comment? (fine to ignore, but might be easy to workaround): https://gist.github.com/benvanik/b690efc4f4320b9d3ea96dd72f633127#file-1-mobilenet_stats-csv

![image](https://user-images.githubusercontent.com/4010439/150875930-edeef628-44e7-4123-9eed-48a470b0b122.png)

Reply:
yeah, they only allow a single header row and we want multiple tables in the output - excel/sheets is fine with it because you select your rows to do analysis/charts/etc on - for gists copying out the single table works (the 0 file in there) and workflow-wise it's like sharing a segment of a print-ir-after-all dump